cpp/source/deal.II/post_processor.h

```diff
@@ -24,14 +24,18 @@ template <int dim>
 class PostprocessorParameters
 {
 public:
-  PostprocessorParameters(std::shared_ptr<boost::property_tree::ptree const> d)
-      : database(d)
+  PostprocessorParameters(
+      std::shared_ptr<boost::property_tree::ptree const> d,
+      std::shared_ptr<dealii::DoFHandler<dim>> const dof_handler)
+      : dof_handler(dof_handler), database(d)
   {
+    BOOST_ASSERT_MSG(dof_handler != nullptr, "Invalid DoFHandler.");
+    BOOST_ASSERT_MSG(database != nullptr, "Invalid database.");
```

Comment:
you know shared_ptr have embedded explicit conversion to bool (see [here](http://www.cplusplus.com/reference/memory/shared_ptr/operator%20bool/)) right?